json_rpc/jsonmarshal.nim

```diff
@@ -94,6 +95,16 @@ proc fromJson*(n: JsonNode, argName: string, result: var uint64) =
       ValueError, "JSON-RPC input is an unexpected negative value")
   result = uint64(asInt)
 
+proc fromJson*(n: JsonNode, argName: string, result: var uint32) =
+  n.kind.expect(JInt, argName)
+  let asInt = n.getInt()
+  # signed -> unsigned conversions are unchecked
+  # https://github.com/nim-lang/RFCs/issues/175
+  if asInt < 0:
+    raise newException(
+      ValueError, "JSON-RPC input is an unexpected negative value")
+  result = uint32(asInt)
```

Comment:
we try to avoid using `result`, it disables several useful compiler diagnostics

Reply:
Makes sense. I'd suggest changing it to `argValue`, but I'm hesitant to do so as it will deviate from the pattern of all the other `fromJson*` in the same file and the[ documentation here](https://github.com/status-im/nim-json-rpc#parameters). Otherwise I could change it for everything else too?

Reply:
right,  I missed that this was a `var` parameter - my comment mainly applies to the magic implicit variable - though reusing the name clearly causes this kind of confusion - it's really an unfortunate feature of nim's - but we can leave it for a separate PR to refactor at some point - there's a lot of legacy code out there that follows practises that have been shown to not work well

Reply:
Ah, great. Thanks, Jacek! Yeah, I've discovered a number of these Nim intricacies, though in general I'm quite enjoying its flexibility.
Can this PR be merged then, or should we log the refactor work first?